doc/userguide/rules/datasets.rst

```diff
@@ -146,6 +165,47 @@ The rules will only match if the data is in the list and the reputation
 value is higher than 200.
 
 
+.. _datasets_json:
+
+dataset with JSON
+~~~~~~~~~~~~~~~~~
+
+Dataset with JSON allows matching data against a set and output data attached to the matching
+value in the event.
+
+There is two format supported: ``json`` and ``ndjson``. The difference is that
+``json`` format is a single JSON object, while ``ndjson`` is handling file with
+one JSON object per line. The ``ndjson`` format is useful for large files
+as the parsing is done line by line.
+
+Syntax::
+
+    dataset:<cmd>,<name>,<options>;
+
+    dataset:<isset|isnotset>,<name> \
+        [, type <string|md5|sha256|ipv4|ip>, load <file name>, format <json|ndjson>, memcap <size>, hashsize <size>, enrichment_key <json_key> \
+         , value_key <json_key>, array_key <json_path>];
+
+Example rules could look like::
+
+    alert http any any -> any any (msg:"IP match"; ip.dst; dataset:isset,bad_ips, type ip, load bad_ips.json, format json, enrichment_key bad_ones, value_key ip; sid:8000001;)
```

Comment:
Could `value_key` simply be `key`? For the JSON format, we'll always need a 'primary' key of sorts, so just simplify it to key.

Reply:
Make sense!